### PR TITLE
[installer-tests] Splitting the integration tests

### PR DESCRIPTION
**Pull request by nandajavarma**
## Description

So far we have been trying to run all the integration tests every night and this caused several timeout issues and we noticed some tests affecting the setup in a way that other tests ended up failing. So this is the first step towards making the tests pass. This PR splits up the tests team wise (IDE, workspaces, and webapp). Every night, against each of the architecture one of the tests run and send slack alert to the corresponding team channel. If one wants to explicitly specify a test suite they can use the annotation `testSuite` with one of these values: `ide`, `workspaces`, `webapp`.

Alongside the PR also increases the timeout of each test to the maximum value. This ensure each test gets as much time as possible and if it get stuck, the process will be killed by the hard timeout of `werft`(2 hours)

## Related Issue(s)

Fixes #

## How to test

You can try running:

```
werft run github -j .werft/gke-installer-tests.yaml
# OR
werft run github -j .werft/gke-installer-tests.yaml -a testSuite=ide
```

See [gitpod-custom-nvn-fix-int-test.10](https://werft.gitpod-dev.com/job/gitpod-custom-nvn-fix-int-test.10)

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by roboquat** [comment]
@nandajavarma: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: .werft/installer-tests.ts [hunk 1]
```diff
@@ -369,10 +376,13 @@ export async function installerTests(config: TestConfig) {
 }
 
 function runIntegrationTests() {
-    werft.phase("run-integration-tests", "Run all existing integration tests");
+    const testComponent: string = randomize(["WORKSPACES", "IDE", "WEBAPP"])
```
Comment (andreafalzetti) [inline .werft/installer-tests.ts:379]: why do we `randomize` here? 🤔

Comment (nandajavarma) [reply .werft/installer-tests.ts:379]: The idea is that since these tests are run every night we want one of the three test suites to run each night. Because running all three tests together tend to hit the 2 hour timeout of werft jobs. So in average over a week, it is likely that all tests will run. We can add an annotation to explicitly specify the test suite, eg: `-a test=IDE`, in which case it will take that as `testComponent` instead of calling the `randomize`

File: .werft/installer-tests.ts [hunk 2]
```diff
@@ -369,10 +376,13 @@ export async function installerTests(config: TestConfig) {
...
+    werft.phase(`run-${testComponent}-integration-tests`, `Run all ${testComponent} integration tests`);
+    console.log(`Running ${testComponent} tests`)
     const slackAlerts = new Map<string, string>([]);
-    for (let test in TESTS) {
-        const testPhase = TESTS[test];
+    const testSuite = eval(`${testComponent}_TESTS`)
```
[the remark this answers is not in the document]
Comment (nandajavarma) [reply .werft/installer-tests.ts:383]: Fixed. Apologies, it is my first time in the javascript world :sweat_smile:

**Comment by nandajavarma** [comment]
@iQQBot @andreafalzetti I have reduced the timeout to 30m. Along with this, it is worth noting that, the only tests that will be run together are vscode IDE test and jetbrains IDE test. Since we consider them part of a suite. They can both take 30 minutes and still wouldn't hit the time limit of the werft job.

File: .werft/installer-tests.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrienthebo) [reply .werft/installer-tests.ts]: No worries! And likewise, apologies if I got jumpy; things like `eval` totally make sense in a lot of other contexts (**cough** bash) because there aren't better tools; while they look reasonable in Javascript/Typescript the risk associated with eval might not be visible until _after_ a CVE comes out. I appreciate you chasing this down! ✨

**Review by adrienthebo — APPROVED** [review]
LGTM!

/hold for Platform review

**Comment by akosyakov** [comment]
> @iQQBot @andreafalzetti I have reduced the timeout to 30m. Along with this, it is worth noting that, the only tests that will be run together are vscode IDE test and jetbrains IDE test. Since we consider them part of a suite. They can both take 30 minutes and still wouldn't hit the time limit of the werft job. You can test it by running: werft run github -j .werft/gke-installer-tests.yaml -a testSuite=ide against this branch

Maybe it make sense to separate them as well? Within IDE team different people are responsible for them and they should be able to rerun independent?

**Comment by nandajavarma** [comment]
@akosyakov That is fair! I will tackle that as a separate PR since we have gone through quite a lot of iterations. /unhold
